lib/matplotlib/dviread.py

```diff
@@ -939,10 +939,16 @@ def _parse_and_cache_line(self, line):
 
         if basename is None:
             basename = tfmname
-        if encodingfile is not None and not encodingfile.startswith(b"/"):
-            encodingfile = find_tex_file(encodingfile)
-        if fontfile is not None and not fontfile.startswith(b"/"):
-            fontfile = find_tex_file(fontfile)
+        if encodingfile is not None:
+            if not encodingfile.startswith(b"/"):
+                encodingfile = find_tex_file(encodingfile)
+            else:
+                encodingfile = encodingfile.decode('utf-8', errors='replace')
```

Comment:
that should be os.fsdecode then?  at least on linux...  (using surrogateescape rather than replace may matter)

Reply:
This matches what `find_tex_file` does.

Reply:
that's what find_tex_file does for `filename` which should effectively be just a filename (not an absolute path); the absolute path's encoding is determined by the kwargs a bit further down (well, things are a bit more complicated, but still...).

Reply:
But that's the encoding for communicating to/from `kpsewhich` only. For converting the input to `find_tex_file` (which is directly from _this_ file), it uses replaced `utf-8` like above.

Reply:
Can we split that out to a separate issue/PR, and keep the type instability for now?  I am not convinced that this is correct, but mostly just need to spend some time setting up a system with weird fsencoding for testing...

Reply:
Sure, can do that.

Reply:
Actually, looking at it again, it seems that we can just remove the `startswith("/")` check and *always* pass things to find_tex_file.  I have checked that kpsewhich (whether called directly or via luatex) will just happily pass-through absolute paths, so we don't need to pre-filter them out.  It is true that in theory this may make things slightly slower (due to the subprocess interaction), but in practice I haven't seen *any* absolute paths in pdftex.map either on my machine or on the shared macos...

Reply:
Hmm, okay, changed to call that always then.